packages/next/pages/_document.tsx

```diff
@@ -84,6 +84,7 @@ function getPreNextScripts(context: HtmlProps, props: OriginProps) {
           key={scriptProps.src || index}
           defer={!disableOptimizedLoading}
           nonce={props.nonce}
+          data-nscript="beforeInteractive"
```

Comment:
Should this be `data-nscript={strategy}`?

Reply:
Also, this doesn't seem right since its not using `next/script` here

Reply:
For `beforeInteractive` we pass the data via context so we can render it during SSR. Only `beforeInteractive` scripts are rendered here